modules/db/src/main/scala/gem/dao/SmartGcalDao.scala

```diff
@@ -1,10 +1,45 @@
 package gem.dao
 
 import gem.config.{F2SmartGcalKey, GcalConfig, SmartGcalKey}
-import gem.enum.{GcalBaselineType, GcalLampType}
+import gem.enum._
 import doobie.imports._
+import scalaz._, Scalaz._
 
 object SmartGcalDao {
+
+  def selectF2(k: F2SmartGcalKey, t: SmartGcalType): ConnectionIO[List[Int]] = {
+    def byLamp(l: GcalLampType): ConnectionIO[List[Int]] =
+      sql"""
+        SELECT gcal_id
+          FROM smart_f2
+         WHERE lamp      = $l :: gcal_lamp_type
+           AND disperser = ${k.disperser}
+           AND filter    = ${k.filter}
+           AND fpu       = ${k.fpu}
+      """.query[Int].list
+
+    def byBaseline(b: GcalBaselineType): ConnectionIO[List[Int]] =
+      sql"""
+        SELECT gcal_id
+          FROM smart_f2
+         WHERE baseline  = $b :: gcal_baseline_type
+           AND disperser = ${k.disperser}
+           AND filter    = ${k.filter}
+           AND fpu       = ${k.fpu}
+      """.query[Int].list
+
+    t.fold(byLamp, byBaseline)
```

Comment:
Here the `byLamp` and `byBaseline` queries are nearly identical, so it's a shame not to abstract this somehow.